src/controllers/rest/impl/AlbumController.ts

```diff
@@ -0,0 +1,227 @@
+    ): Promise<AlbumModel> {
+        return this.albumService.disassociateFilesFromAlbum(albumToken, fileTokens);
+    }
+
+    @Delete("/:albumToken")
+    @Returns(StatusCodes.OK, SuccessModel)
+    @Returns(StatusCodes.BAD_REQUEST, DefaultRenderException)
+    @Description("Delete an album")
+    @Summary("Delete album")
+    public async deleteAlbum(
+        @Description("The album token to associate the file with")
+        @PathParams("albumToken")
+        albumToken: string,
+        @Description("Delete files, if false then the files will remain in the bucket")
+        @QueryParams("deleteFiles")
+        @Default(false)
+        @Optional()
+        deleteFiles: boolean,
+        @Res() res: PlatformResponse,
+    ): Promise<PlatformResponse> {
+        await this.albumService.deleteAlbum(albumToken, deleteFiles);
+        return super.doSuccess(res, "album deleted");
+    }
+
+    @Get("/:albumToken")
+    @Returns(StatusCodes.OK, AlbumDto)
+    @Returns(StatusCodes.BAD_REQUEST, DefaultRenderException)
+    @Returns(StatusCodes.NOT_FOUND, DefaultRenderException)
+    @Description("Get an album and all files associated with it")
+    @Summary("Get full album")
+    public async getAlbum(
+        @Description("The album to get")
+        @PathParams("albumToken")
+        albumToken: string,
+    ): Promise<AlbumModel> {
+        const album = await this.albumService.getAlbum(albumToken);
+        if (album.isPublicToken(albumToken)) {
+            throw new BadRequest("Supplied token is not valid");
+        }
+        return album;
+    }
+
+    @Get("/public/:albumToken")
+    @Returns(StatusCodes.OK, PublicAlbumDto)
+    @Returns(StatusCodes.BAD_REQUEST, DefaultRenderException)
+    @Returns(StatusCodes.NOT_FOUND, DefaultRenderException)
+    @Description("Get the public sharable view of an album")
+    @Summary("Get an album in public view")
+    public async getPublicAlbum(
+        @Description("The album to get, this should be the public token")
+        @PathParams("albumToken")
+        albumToken: string,
+    ): Promise<AlbumModel> {
+        const album = await this.albumService.getAlbum(albumToken);
+        if (!album.isShared) {
+            throw new BadRequest("This album is not public");
+        }
+        if (!album.isPublicToken(albumToken)) {
+            throw new BadRequest("Supplied token is not valid");
+        }
+        return album;
+    }
+
+    @Get("/share/:albumToken")
+    @(Returns(StatusCodes.OK, SuccessModel).Description("description will contain the URL"))
+    @Returns(StatusCodes.BAD_REQUEST, DefaultRenderException)
+    @Returns(StatusCodes.NOT_FOUND, DefaultRenderException)
+    @Description("Share album, this returns a public URL to the album")
+    @Summary("Share an album")
+    public async shareAlbum(
+        @Description("The private token to the album")
+        @Required()
+        @PathParams("albumToken")
+        albumToken: string,
+        @Res() res: PlatformResponse,
```

Comment:
Avoid Res injection. you can deal with `InjectContext` annotation to retrieve res/req. I know, doSucces come from the base class and it seems a huge refactoring here.